bson/doc.go

```diff
@@ -19,24 +20,96 @@
 // 		i32, ok := val.Int32OK()
 // 		// do something with i32...
 //
-// The D family of types is used to build concise representations of BSON using native Go types.
-// These types do not support automatic lookup.
+// Native Go Types
+//
+// This D and M types defined in this package can be used to build concise representations of BSON using native Go
```

Comment:
By concise, do you mean that the code is concise or the representation in memory is concise?  Likely the former, but it reads like you mean the latter.  Consider saying directly that M is a map and D is a slice.

Reply:
It's supposed to mean that the code is concise, but I can see how that's confusing. Changed to say that M is a map and D is a slice.